### PR TITLE
Adds pipenet collectors

### PR DESCRIPTION
**Pull request by Andromeda-K22**
🆑 
Adds pipenet enable collectors. Position them over a pipe cap, and wrench into place.
Uses phoron from the pipenet instead of a tank. Adminspawn only currently.
🆑

### CODE REVIEW

**Comment by Andromeda-K22** [comment]
Not my runtime. Re-run travis pls.

File: code/modules/power/singularity/collector.dm [hunk 1]
```diff
@@ -163,3 +163,101 @@ var/global/list/rad_collectors = list()
...
+	var/obj/machinery/atmospherics/pipe/cap/pipe = null
+	var/datum/gas_mixture/removed = null
+
+/obj/machinery/power/rad_collector/pipenet/New()
+	..()
+	rad_collectors += src
+	if(anchored)
+		locate_pipenet()
+
+/obj/machinery/power/rad_collector/pipenet/Destroy()
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]: Needs to handle destruction of the pipe reference and the gas mixture.

File: code/modules/power/singularity/collector.dm [hunk 2]
```diff
@@ -163,3 +163,101 @@ var/global/list/rad_collectors = list()
...
+				src.locked = 0 //just in case it somehow gets locked
+				to_chat(user, "<span class='warning'>The controls can only be locked when the [src] is active</span>")
+		else
+			to_chat(user, "<span class='warning'>Access denied!</span>")
+		return 1
+	return ..()
+
+
+/obj/machinery/power/rad_collector/pipenet/proc/locate_pipenet()
+	var/T = src.loc
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]:
```suggestion
	var/turf/T = get_turf(src)
```

File: code/modules/power/singularity/collector.dm [hunk 3]
```diff
@@ -163,3 +163,101 @@ var/global/list/rad_collectors = list()
...
+			"You [anchored? "secure":"undo"] the external bolts.", \
+			"You hear a ratchet")
+		if(anchored)
+			connect_to_network()
+			locate_pipenet()
+		else
+			disconnect_from_network()
+			P = null
+		return 1
+	else if(istype(W, /obj/item/weapon/card/id)||istype(W, /obj/item/modular_computer))
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]: You can just let `return ..()` handle this.

File: code/modules/power/singularity/collector.dm [hunk 4]
```diff
@@ -163,3 +163,101 @@ var/global/list/rad_collectors = list()
...
+
+/obj/machinery/power/rad_collector/pipenet
+	name = "Modular Radiation Collector Array"
+	desc = "A variant of the radiation collector array which connects to a pipecap underneath it."
+	var/obj/machinery/atmospherics/pipe/cap/pipe = null
+	var/datum/gas_mixture/removed = null
+
+/obj/machinery/power/rad_collector/pipenet/New()
+	..()
+	rad_collectors += src
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]: Left over from the parent proc.

File: code/modules/power/singularity/collector.dm [hunk 5]
```diff
@@ -163,3 +163,101 @@ var/global/list/rad_collectors = list()
...
+	if(anchored)
+		locate_pipenet()
+
+/obj/machinery/power/rad_collector/pipenet/Destroy()
+	rad_collectors -= src
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]: Same as above. It'll already be removed in the parent call.

File: code/modules/power/singularity/collector.dm [hunk 6]
```diff
@@ -163,3 +163,85 @@ var/global/list/rad_collectors = list()
...
+/obj/machinery/power/rad_collector/pipenet
+	name = "Modular Radiation Collector Array"
+	desc = "A variant of the radiation collector array which connects to a pipecap underneath it."
+	var/obj/machinery/atmospherics/pipe/cap/pipe = null
+	var/datum/gas_mixture/removed = null
+
+/obj/machinery/power/rad_collector/pipenet/Destroy()
+	QDEL_NULL(removed)
+	pipe = null
+	..()
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]: Missed this, all Destroy()'s need to return their hint.
```suggestion
	. = ..()
```

File: code/modules/power/singularity/collector.dm [hunk 7]
```diff
@@ -163,3 +163,85 @@ var/global/list/rad_collectors = list()
 		flick("ca_deactive", src)
 	update_icons()
 	return
+
+/obj/machinery/power/rad_collector/pipenet
+	name = "Modular Radiation Collector Array"
+	desc = "A variant of the radiation collector array which connects to a pipecap underneath it."
+	var/obj/machinery/atmospherics/pipe/cap/pipe = null
+	var/datum/gas_mixture/removed = null
+
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]: You went a little bit overkill and removed the connecting to a pipenet in New() part. Also you can just stuff it in Initialize instead.

File: code/modules/power/singularity/collector.dm [hunk 8]
```diff
@@ -163,3 +163,90 @@ var/global/list/rad_collectors = list()
...
+/obj/machinery/power/rad_collector/pipenet/Destroy()
+	QDEL_NULL(removed)
+	pipe = null
+	. = ..()
+
+/obj/machinery/power/rad_collector/pipenet/Initialize()
+	if(anchored)
+		locate_pipenet()
+		connect_to_network()
+
```
Comment (Naxodile) [inline code/modules/power/singularity/collector.dm]: It's that time again. All Initialize()'s must return their parent's init hint or a custom one.

**Pull request merged**